torch/csrc/lazy/core/shape.h

```diff
@@ -17,24 +18,60 @@ class TORCH_API Shape {
 
   std::string to_string() const;
 
-  c10::ScalarType scalar_type() const { return scalar_type_; }
-  void set_scalar_type(at::ScalarType value) { scalar_type_ = value; }
+  c10::ScalarType scalar_type() const {
+    return scalar_type_;
+  }
+  void set_scalar_type(at::ScalarType value) {
+    scalar_type_ = value;
+  }
+
+  int64_t dim() const {
+    return sizes_.size();
+  }
+  c10::ArrayRef<int64_t> sizes() const {
+    return sizes_;
+  }
+  int64_t size(int64_t dim) const {
+    return sizes_.at(dim);
+  }
+  void set_size(int64_t dim, int64_t size) {
+    sizes_.at(dim) = size;
+  }
+
+  const at::optional<std::vector<bool>>& is_symbolic() const {
+    return is_symbolic_;
+  }
+  c10::SymbolicShape get_symbolic_shape() const;
```

Comment:
I'm 100% sure XLA wants to know about yet another abstraction? can we just expose `is_dynamic(X)` ?